src/E2_economics.py

```diff
@@ -57,138 +64,401 @@
+    either the total lifetime investment costs or the upfront investment costs are calculated,
+
+    Parameters
+    ----------
+    specific_cost: float
+        a) Specific per-unit investment costs of an asset over its lifetime, including all replacement costs
+        b) Specific per-unit investment costs of an asset in year 0
+
+    capacity: float
+        Capacity to be installed
+
+    development_costs: float
+        Fix development costs, ie. an expense not related to the capacity that is installed. Could be planning costs of the plant.
+
+    Returns
+    -------
+    costs_investment: float
+        a) Total investment costs of an asset over its lifetime, including all replacement costs
+        b) Upfront investment costs in year 0
+
+    """
+    costs_upfront_investment = specific_cost * capacity + development_costs
+    return costs_upfront_investment
+
+
+def calculate_total_capital_costs(upfront, replacement):
+    r"""
+    Calculated total capital expenditures
+
+    Parameters
+    ----------
+    upfront: float
+        Upfront investments at t=0, including development costs
+
+    replacement: float
+        Replacement costs of pre-installed and new assets
+
+
+    Returns
+    -------
+    cost_total_investment: float
+        Total capital costs
+    """
+    cost_total_investment = upfront + replacement
+    return cost_total_investment
+
+
+def calculate_costs_replacement(
+    specific_replacement_of_initial_capacity,
+    specific_replacement_of_optimized_capacity,
+    initial_capacity,
+    optimized_capacity,
+):
+    r"""
+    Calculates (the present value of) the replacement costs over the project lifetime
+
+    Parameters
+    ----------
+    specific_replacement_of_initial_capacity: float
+        Per-unit replacement costs of an asset that was pre-existing at the location
+
+    specific_replacement_of_optimized_capacity: float
+        Per-unit replacement costs of an asset that is to be installed
+
+    initial_capacity: float
+        Initial capacity installed
+
+    optimized_capacity: float
+        Add capacity to be installed, as optimized
+
+    Returns
+    -------
+    costs_replacements: float
+        Aggregated replacement costs over the project lifetime
+    """
+    costs_replacements = (
+        specific_replacement_of_initial_capacity * initial_capacity
+        + specific_replacement_of_optimized_capacity * optimized_capacity
+    )
+    return costs_replacements
+
+
+def calculate_operation_and_management_expenditures(
+    specific_om_cost, installed_capacity, optimized_add_capacity
+):
+    r"""
+    Calculates operation and management expenditures
+
+    Parameters
+    ----------
+    specific_om_cost: float
+        a) specific operation and management costs per unit in year 0
+        b) specific operation and management costs per unit for the whole project lifetime
+
+    installed_capacity: float
+        Capacity installed initially
+
+    optimized_add_capacity: float
```

Comment:
is this similar to `optimized_capacity` in the function above? If yes, why add the `_add`, `optimized_capacity` seems like a good name :)

Reply:
I wanted to make the distinction between installed cap and optimized *additional* cap clearer here.